esprima.js

```diff
@@ -674,6 +677,19 @@
 
         // Check for most common single-character punctuators.
         case 0x2E:  // . dot
+            ch3 = source.substr(index, 3);
```

Comment:
While this might look like a micro-optimization, I think we should consider checking for another dot before making `substr` call. For example, in a library as large as jQuery, there could > 5000 hits to this place. Since it follows that `...` is not widely used, it is safe to assume that we need to fail fast in that case, without incurring the function call overhead.

Also, it seems like an arbitrary choice, but I think falling through here it's not necessary (i.e. just return the dot token). I am ambivalent about each variant, although it's usually safe to err on the not-falling-through case.

Having said that, I think it's fine to address that minor tweak in a separate follow-up.